Gems/AssetValidation/Code/Tests/AssetValidationTestShared.h

```diff
@@ -19,180 +19,181 @@
+        : UnitTest::ScopedAllocatorSetupFixture
+        , UnitTest::SetRestoreFileIOBaseRAII
+        , AzFramework::ApplicationRequests::Bus::Handler
     {
-        AZ_Assert(false, "Not implemented");
-    }
+        AssetValidationTest()
+            : UnitTest::SetRestoreFileIOBaseRAII(m_fileIO)
+        {
+            if (!AZ::SettingsRegistry::Get())
+            {
+                AZ::SettingsRegistry::Register(&m_registry);
+
+                auto projectPathKey = AZ::SettingsRegistryInterface::FixedValueString(AZ::SettingsRegistryMergeUtils::BootstrapSettingsRootKey)
+                    + "/project_path";
+                m_registry.Set(projectPathKey, (AZ::IO::FixedMaxPath(m_tempDir.GetDirectory()) / "AutomatedTesting").Native());
+                AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddRuntimeFilePaths(m_registry);
+
+                // Set the engine root to the temporary directory and re-update the runtime file paths
+                auto enginePathKey = AZ::SettingsRegistryInterface::FixedValueString(AZ::SettingsRegistryMergeUtils::BootstrapSettingsRootKey)
+                    + "/engine_path";
+                m_registry.Set(enginePathKey, m_tempDir.GetDirectory());
+                AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddRuntimeFilePaths(m_registry);
+            }
+        }
 
-    void CalculateBranchTokenForEngineRoot([[maybe_unused]] AZStd::string& token) const override
-    {
-        AZ_Assert(false, "Not implemented");
-    }
+        void NormalizePath(AZStd::string&) override
+        {
+            AZ_Assert(false, "Not implemented");
+        }
 
-    void SetUp() override
-    {
-        using namespace ::testing;
+        void NormalizePathKeepCase(AZStd::string&) override
+        {
+            AZ_Assert(false, "Not implemented");
+        }
 
-        ASSERT_TRUE(m_tempDir.IsValid());
+        void CalculateBranchTokenForEngineRoot([[maybe_unused]] AZStd::string& token) const override
+        {
+            AZ_Assert(false, "Not implemented");
+        }
 
-        AzFramework::ApplicationRequests::Bus::Handler::BusConnect();
-    }
+        void SetUp() override
+        {
+            AzFramework::ApplicationRequests::Bus::Handler::BusConnect();
+        }
 
-    void TearDown() override
-    {
-        AzFramework::ApplicationRequests::Bus::Handler::BusDisconnect();
-    }
+        void TearDown() override
+        {
+            AzFramework::ApplicationRequests::Bus::Handler::BusDisconnect();
+        }
 
-    bool CreateDummyFile(const char* path, const char* seedFileName, AZStd::string_view contents, AZStd::string& subfolderPath) const;
+        bool CreateDummyFile(const AZ::IO::Path& path, AZStd::string_view contents) const;
 
-    AZ::IO::LocalFileIO m_fileIO;
-    UnitTest::ScopedTemporaryDirectory m_tempDir;
-    AZ::SettingsRegistryImpl m_registry;
-};
+        AZ::IO::LocalFileIO m_fileIO;
+        AZ::Test::ScopedAutoTempDirectory m_tempDir;
```

Comment:
This is the only meaningful change in this file.

Everything else is just indentation due to move the contents under a namespace

Reply:
Thank you for mentioning that.  This diff chunking algo is bonkers.